applications/informationmodel-cat/src/main/java/no/fdk/imcat/controller/InformationModelSearchController.java

```diff
@@ -0,0 +1,170 @@
+
+        if (!orgPath.isEmpty()) {
+            composedQuery.filter(QueryUtil.createTermFilter("publisher.orgPath", orgPath));
+        }
+
+        NativeSearchQuery finalQuery = new NativeSearchQueryBuilder()
+            .withQuery(composedQuery)
+            .withIndices("imcat").withTypes("informationmodel")
+            .withPageable(new PageRequest(from, size))
+            .build();
+
+        if (!StringUtils.isEmpty(returnFields)) {
+            SourceFilter sourceFilter = new FetchSourceFilter(returnFields.concat(",prefLabel").split(","), null);
+            finalQuery.addSourceFilter(sourceFilter);//TODO: Understand
+        }
+
+        if (!StringUtils.isEmpty(sortfield)) {
+            addSort(sortfield, sortdirection, finalQuery);
+        }
+
+        AggregatedPage<InformationModel> aggregatedPage = elasticsearchTemplate.queryForPage(finalQuery, InformationModel.class);
+
+        List<InformationModel> informationModels = aggregatedPage.getContent();
+
+        PagedResources.PageMetadata pageMetadata = new PagedResources.PageMetadata(
+            pageable.getPageSize(),
+            pageable.getPageNumber(),
+            aggregatedPage.getTotalElements(),
+            aggregatedPage.getTotalPages()
+        );
+
+        //In order to not experience double escaping of the schemas, we need to annotate a field with @JsonRawValue, but this only works on output.
+        //So we must have a separate class with the @JsonRawValue field.
+        List<InformationModelForOutput> informationModelForOutput = new ArrayList<>();
+        for (InformationModel model : informationModels) {
+            InformationModelForOutput modelForOutput = new InformationModelForOutput();
+            modelForOutput.setId(model.getId());
+            modelForOutput.setPublisher(model.getPublisher());
+            modelForOutput.setHarvestSourceUri(model.getHarvestSourceUri());
+            modelForOutput.setHarvest(model.getHarvest());
+            modelForOutput.setTitle(model.getTitle());
+            modelForOutput.theSchema = model.getSchema();
+            informationModelForOutput.add(modelForOutput);
+        }
+
+        //PagedResources<InformationModel> InformationModelResources = new PagedResources<>(informationModels, pageMetadata);
+
+        PagedResources<InformationModelForOutput> InformationModeConvertedResources = new PagedResources<>(informationModelForOutput, pageMetadata);
+
+
+        return InformationModeConvertedResources;
+    }
+
+    private void addSort(String sortfield, String sortdirection, NativeSearchQuery searchBuilder) {
+        if (!sortfield.trim().isEmpty()) {
+
+            Sort.Direction sortOrder = sortdirection.toLowerCase().contains("asc".toLowerCase()) ? Sort.Direction.ASC : Sort.Direction.DESC;
+            StringBuilder sbSortField = new StringBuilder();
+
+            if (!sortfield.equals("modified")) {
+                sbSortField.append(sortfield).append(".nb");
```

Comment:
det finnes bare to muligheter, "modified" og "relevance", og relefance er default, så bare modified trenger å være håndtert.

Reply:
ok